csharp/BufferedReader.cs

```diff
@@ -9,6 +9,8 @@ internal sealed class BufferedReader<TPhysicalvalue> where TPhysicalvalue : unma
     {
         public BufferedReader(ColumnReader reader, TPhysicalvalue[] values, short[]? defLevels, short[]? repLevels)
         {
+            if (defLevels == null) throw new InvalidOperationException("definition levels not defined");
```

Comment:
Ah I see BufferedReader is only used for reading arrays now. So we would expect repLevels to always be present too right? Also, would it be valid to have required arrays such that defLevels were not present?

If we are going to do a null check here then the private properties should probably be non-nullable so that the `?? -1` isn't required below.

Reply:
I had a play around with this to try to understand things better. In PyArrow you can round-trip a required list, but we don't currently support reading this in ParquetSharp and crash with `elementType is an array but schema does not match the expected layout`. But you do always have defLevels set even for required lists due to the middle repeated "list" schema node. (would probably be nice to fix this later but not high priority)

And I see now that for nested fields without arrays, repLevels will be null, so that needs to stay as optional.

For non-nullable nested fields defLevels may not be defined though, so I think this null check is actually incorrect. However if I comment it out I get an "Invalid input stream" exception later due to checking `defn.DefLevel < nullDefinitionLevel` in `MakeLeafReaderSingle` where `defn.DefLevel == -1`. Eg. I can't currently read a file generated like this:
```python
schema = pa.schema([pa.field('struct', type=pa.struct([pa.field('el', pa.int32(), nullable=False)]), nullable=False)])
table = pa.Table.from_arrays([[(1, ), (2, ), (3, )]], schema=schema)
pq.write_table(table, 'test.parquet')
```
It looks like the fix is as simple as changing the `-1`s below to `0`, which makes more sense to me; if there are no definition levels required then everything must be defined at level 0 (and probably similarly for repLevels?).